e2e/test/helpers/TestDevice.cs

```diff
@@ -140,32 +81,70 @@ public string ConnectionString
         /// </summary>
         public Device Device { get; private set; }
 
+        public IotHubDeviceClient DeviceClient { get; private set; }
+
         public IAuthenticationMethod AuthenticationMethod { get; private set; }
 
-        public IotHubDeviceClient CreateDeviceClient(IotHubClientOptions options = default, ConnectionStringAuthScope authScope = ConnectionStringAuthScope.Device)
+        public async Task<IotHubDeviceClient> CreateDeviceClientAsync(IotHubClientOptions options = default, ConnectionStringAuthScope authScope = ConnectionStringAuthScope.Device, bool openClient = false)
```

Comment:
nit: should we rename this so that the method name calls out that the returned client might optionally be already open?